src/TRestGeant4Metadata.cxx

```diff
@@ -1172,6 +1182,102 @@ void TRestGeant4Metadata::PrintMetadata() {
     metadata << endl;
 }
 
+///////////////////////////////////////////////
+/// \brief Reads an input file produced by restGenerator.
+///
+///
+void TRestGeant4Metadata::ReadGeneratorTreeFile(TString fName) {
+    debug << "TRestGeant4Metadata::ReadGeneratorTreeFile" << endl;
+
+    TFile* file;
+    TTree* GeneratorTree;
+    TString treeName = "GeneratorTree";
+
+    file = TFile::Open(fName, "READ");
+    if (!file) {
+        ferr << "Error opening file: " << fName << endl;
+        exit(1);
+    }
+    file->GetObject(treeName, GeneratorTree);
+
+    if (!GeneratorTree) {
+        ferr << "Error reading TTree named " << treeName << " from " << fName << endl;
+        file->ls();
+        exit(1);
+    }
+
+    int numberOfGeneratorEvents = GeneratorTree->GetEntries();
+    if (numberOfGeneratorEvents <= 0) {
+        ferr << "Generator TTree: " << treeName << " from " << fName << " has no events" << endl;
+        exit(1);
+    }
+
+    debug << "Reading generator file : " << fName << endl;
+    debug << "Total number of events in TTree : " << numberOfGeneratorEvents << endl;
+    /*
+    fNEvents = numberOfGeneratorEvents;
+    warning << "Overriding NEvents parameter with value " << fNEvents << " to new value "
+            << numberOfGeneratorEvents << endl;
+    */
+    debug << "Setting up TTree" << endl;
+
+    vector<double>*x = 0, *y = 0, *z = 0;
+    vector<double>*px = 0, *py = 0, *pz = 0;
+    vector<double>*KE = 0, *timeGlobal = 0, *weight = 0;
+    vector<TString>* particleName = 0;
```

Comment:
What is the reason of defining a pointer to a vector? You should intialize as nullptr anyhow